src/care/LoopFuser.h

```diff
@@ -211,17 +210,17 @@ __global__ void write_launcher_ptr(basil::detail::device_wrapper_ptr* out)
 template<typename ReturnType, typename kernel_type>
 inline void * get_launcher_wrapper_ptr(bool get_if_null)
 {
-#if defined __GPUCC__ && defined GPU_ACTIVE
+#if defined CARE_GPUCC && defined GPU_ACTIVE
    static_assert(alignof(kernel_type) <= sizeof(basil::detail::device_wrapper_ptr),
                  "kernel_type has excessive alignment requirements");
    static basil::detail::device_wrapper_ptr ptr = nullptr;
    if (ptr == nullptr && get_if_null) {
       basil::detail::device_wrapper_ptr* pinned_buf = basil::detail::get_pinned_device_wrapper_ptr_buf();
-      cudaStream_t stream = 0;
+      gpuStream_t stream = 0;
       void* func = (void*)&write_launcher_ptr<ReturnType, kernel_type>;
       void* args[] = { &pinned_buf };
-      BASIL_cudaCheck(cudaLaunchKernel(func, 1, 1, args, 0, stream));
-      BASIL_cudaCheck(cudaStreamSynchronize(stream));
+      care::gpuLaunchKernel(func, 1, 1, args, 0, stream);
```

Comment:
Does gpuLaunchKernel check the return value for errors?

Reply:
Yes. I moved the check for errors within the functions for all of these. That way, nobody will ever forget to check the error code; always done for us without adding extra cruft around each function call